### ISSUE TITLE
Extreme lag/freeze with BW

### ISSUE THREAD

**Issue by zofia1347** [SLOKnightfall/BetterWardrobe#350]
I don't have any sort of LUA errors but for whatever reason, I realized after some addon troubleshooting that BetterWardrobe and Bagnon seem to have some sort of confliction? BetterWardrobe freezes up my entire game so bad (especially if I open up during combat while checking tmogs) with Bagnon enabled and all of my other addons still on but the moment Bagnon is turned off it works just fine! 

Been updating both mods pretty regularly when updates come out and no luck. BW and Bagnon have worked just fine in the past expansions for me with but since Dragonflight it's been an issue. I wish I had more helpful information to give -- let me know if I can provide anything else..? I apologize for the vague "it's not working!!" post lol.

**Comment by n1ckal** [comment]
Same as issue #338 , and now #352

**Comment by SLOKnightfall** [comment]
Well it took a while but I finally managed to track down the cause of the lag when Bagnon is enabled.   Bagnon hooks into the TooltipDataProcessor, so every time a tooltip is shown Bagnon tries to look up how many of the item a player has and adds that info to the tooltip.  BetterWardrobe uses hidden tooltips to look up class data for the sets window.  So when scrolling Bagnon is queuing up a bunch of lookups which is causing the lag.   A partial fix is that I've added a cache for the data from the tooltips, so after the initial lookup BetterWardrobe can use the cache eliminating Bagnon from triggering multiple lookups and causing some lag.  Unfortunately there is still going to be some initial lag, but once things get caches the lag should stop.  Additionally I found a few places where things were not getting cached properly and fixed them, so that hopefully should clear up some of the non Bagnon lag.
